### ISSUE TITLE
what is the difference to quibble?

### ISSUE THREAD

**Issue by TimDaub** [iambumblehead/esmock#11]
see: https://github.com/testdouble/quibble

It seems that the functionality is the same, isn't it?

**Comment by iambumblehead** [comment]
@TimDaub esmock uses separated mock definitions for each module and I don't think quibble does that. One process can run simultaneous tests importing the same module but with different mock definitions. Esmock locates the original module from the filesystem and maps mocked values into the originally exported ones and quibble's author thinks that is a bad practice.

There could be other differences but I'm not familiar with quibble since researching it several months ago.

**Comment by TimDaub** [comment]
With quibble, I ran into issues where in ava test - if I used quibble to mock a module in one test of a file, but then used that module in another test too - it'd now allow me to run those tests in parallel. Instead the program would start throwing errors as the mocking isn't isolated to one test but needs to be reset with `quibble.reset`.

Am I understanding your comment correctly that esmock fixes this problem?

**Comment by iambumblehead** [comment]
@TimDaub yes. And esmock will not have that problem. I believed quibble would have problems in that situation but haven't tested it myself. Thanks for sharing :) I'm glad to know about it.

**Comment by iambumblehead** [comment]
my earlier comment isn't clear and maybe this example is better
```javascript
const mockedInst1 = await esmock( '../src/test.js', { '../src/util.js': { isHappy: () => true } } );
const mockedInst2 = await esmock( '../src/test.js', { '../src/util.js': { isHappy: () => false } } );
```
mockedInst1 and mockedInst2 are separated instances with separately imported dependencies, even though their origin file is same.
